watcher/watcher.go

```diff
@@ -31,6 +34,87 @@ type WatchHandler struct {
 	currentPodListResourceVersion     string // current PodList version, used by watcher (https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes)
 }
 
+// remove unused imageIDs and instanceIDs from storage. Update internal maps
+func (wh *WatchHandler) cleanUp(ctx context.Context) error {
+	storageImageIDs, err := wh.ListImageIDsFromStorage()
+	if err != nil {
+		logger.L().Ctx(ctx).Error("error to ListImageIDsFromStorage", helpers.Error(err))
+	}
+
+	storageInstanceIDs, err := wh.ListInstanceIDsFromStorage()
+	if err != nil {
+		logger.L().Ctx(ctx).Error("error to ListInstanceIDsFromStorage", helpers.Error(err))
+	}
+
+	// list Pods, extract their imageIDs and instanceIDs
+	podsList, err := wh.k8sAPI.ListPods("", map[string]string{})
+	if err != nil {
+		return err
+	}
+
+	// reset maps
+	wh.cleanUpMaps()
+	currentInstanceIDs := wh.getInstanceIDsAndBuildMaps(ctx, podsList)
+
+	// image IDs in storage that are not in current map should be deleted
+	imageIDsForDeletion := make([]string, 0)
+	for _, imageID := range storageImageIDs {
+		if _, ok := wh.imagesIDToWlidsMap[imageID]; !ok {
```

Comment:
[Question]: You don't need to synchronize read/write access to the `wh.imagesIDToWlidsMap` object?

Reply:
yes